### PR TITLE
Adding CodeQL custom queries and scripts

### PR DESCRIPTION
**Pull request by mrragava**
This PR has custom CodeQL and scripts implemented specific to OpenEncalve to test against security vulnerabilities.
These queries are can be used for both CI/CD code scanning and developer analysis using Visual Studio code.

Signed-off-by: mrragava <mrragava@gmail.com>

### CODE REVIEW

File: src/static/codeql/queries/cpp/openenclave/Dereferences.qll
Comment (anakrish) [inline src/static/codeql/queries/cpp/openenclave/Dereferences.qll:43]: OE has its own C library. How are calls to OE's C library functions treated.

File: src/static/codeql/queries/cpp/openenclave/Dereferences.qll
Comment (anakrish) [inline src/static/codeql/queries/cpp/openenclave/Dereferences.qll:67]: Does this handle dataflow:
```
int func(int* a)
{
    int loc_arr[] = { a };
    return *loc_arr[0];
}
```

File: src/static/codeql/queries/cpp/openenclave/ECall-Args-IsOutsideEnclave.ql
Comment (anakrish) [inline src/static/codeql/queries/cpp/openenclave/ECall-Args-IsOutsideEnclave.ql:27]: Don't we need to look at the second argument (the size) as well?

File: src/static/codeql/queries/cpp/openenclave/ECall-Args-TOCTOU.ql
Comment (anakrish) [inline src/static/codeql/queries/cpp/openenclave/ECall-Args-TOCTOU.ql:21]: We might need to extend this check for nested host pointers.

File: src/static/codeql/queries/cpp/openenclave/Memory.qll
Comment (anakrish) [inline src/static/codeql/queries/cpp/openenclave/Memory.qll:12]: There is also oe_memset.

File: src/static/codeql/queries/cpp/openenclave/Memory.qll
Comment (anakrish) [inline src/static/codeql/queries/cpp/openenclave/Memory.qll:26]: Would this handle indirect calls?

File: src/static/codeql/queries/cpp/openenclave/OCall-Args-StructPadding-InfoDisco.ql
Comment (anakrish) [inline src/static/codeql/queries/cpp/openenclave/OCall-Args-StructPadding-InfoDisco.ql:35]: What is the value of checking that the variable is never freed?

File: src/static/codeql/queries/cpp/openenclave/Padding.qll
Comment (anakrish) [inline src/static/codeql/queries/cpp/openenclave/Padding.qll:29]: What does this check do? Stack allocated variables returned?

File: src/static/codeql/queries/cpp/openenclave/OCall-Args-StructPadding-InfoDisco.ql
Comment (anakrish) [inline src/static/codeql/queries/cpp/openenclave/OCall-Args-StructPadding-InfoDisco.ql:28]: Why is initial padding important? Why not padding in middle and at the end?

File: .gitmodules
Comment (anakrish) [inline .gitmodules:11]: oeedger8r is also a submodule of openenclave. There is also the option of running codeql on that submodule instead of directly  on the repo.
Is there a benefit to running codeql directly on the oeedger8r repo?

**Review by anakrish — APPROVED** [review]
Looks great!

**Comment by ghost** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/openenclave/openenclave-security/pull/4#pullrequestreview-582880572" title="Approved">anakrish</a>*, *<a href="https://github.com/openenclave/openenclave-security/pull/4#" title="Author self-approved">mrragava</a>*

The full list of commands accepted by this bot can be found [here](?repo=openenclave%2Fopenenclave-security).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/openenclave/openenclave-security/blob/main/OWNERS)~~ [mrragava]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
